flake.nix

```diff
@@ -2,48 +2,90 @@
   description = "declarative and reproducible Jupyter environments - powered by Nix";
 
   inputs = {
-    flake-utils.url = "github:numtide/flake-utils";
-    nixpkgs.url = "nixpkgs/a5d03577f0161c8a6e713b928ca44d9b3feb2c37";
+    flake-utils.url = github:numtide/flake-utils;
+    flake-compat = {
+      url = github:/teto/flake-compat/support-packages;
+      flake = false;
+    };
+    nixpkgs.url = "github:nixos/nixpkgs/release-21.11";
+    ihaskell.url = github:gibiansky/IHaskell;
   };
 
   outputs =
     inputs@{ self
     , nixpkgs
+    , ihaskell
     , flake-utils
+    , ...
     }:
-    (flake-utils.lib.eachSystem ["x86_64-linux"]
+    (flake-utils.lib.eachSystem ["x86_64-linux" "x86_64-darwin"]
       (system:
       let
         pkgs = import nixpkgs
           {
             inherit system;
+            allowUnsupportedSystem = true;
             overlays = nixpkgs.lib.attrValues self.overlays;
+              # [ self.overlays.jupyterWith ];
           };
+        ihaskellOverlay = ihaskell.packages.${system}.ihaskell-env.ihaskellOverlay;
+
       in
-      rec {
+      {
 
-        defaultPackage = pkgs.jupyterWith.jupyterlabWith {
-          kernels = [];
-        };
+        packages = let
+          iPython = pkgs.jupyterWith.kernels.iPythonWith {
+            name = "Python-data-env";
+            ignoreCollisions = true;
+          };
 
-        lib = {
-          inherit (pkgs.jupyterWith)
-            jupyterlabWith
-            kernels
-            mkBuildExtension
-            mkDirectoryWith
-            mkDirectoryFromLockFile
-            mkDockerImage
-            ;
+          iHaskell = pkgs.jupyterWith.kernels.iHaskellWith {
+            name = "ihaskell-flake";
+            packages = p: with p; [ vector aeson ];
+            extraIHaskellFlags = "--codemirror Haskell"; # for jupyterlab syntax highlighting
+            haskellPackages = pkgs.haskellPackages;
+          };
+
+        in
+        {
+          jupyterEnvironment = pkgs.jupyterWith.jupyterlabWith {
+              kernels = [ iPython iHaskell ];
+              directory = "./.jupyterlab";
```

Comment:
removing this line makes it possible to find back master's behavior. Remove this and then it's ready for merge I think